### DISCUSSION TITLE
Meanweel NPB-750-24

### DISCUSSION THREAD

**Discussion by iroc-fabi** [skippermeister/OpenDTU-OnBattery#8] [Q&A]
Ich habe mir heute das Ladegerät von Meanweel bestellt. 
Ich nutze das Fusionboard mit der Zusatzplatine mit CAN und VE.Direct Anschlüssen. 
Kann ich das Ladegerät einfach anschließen und es wird mit dieser openDTU Version erkannt?

Welche Parameter werden bei der getesteten 48V Version dargestellt? Hat jemand einen Screenshot?

**Comment by iroc-fabi** [comment]
Die Version für das Fusionboard lässt sich leider nicht installieren. 
![IMG_7978](https://github.com/skippermeister/OpenDTU-OnBattery/assets/3811704/77454225-5b94-4ace-8b66-09facc2770b2)

**Reply by sentinel1111** [reply]
Ich habe kein FusionBoard,aber bei mir war es auch so.Ich habe es dann über usb gemacht.

**Reply by iroc-fabi** [reply]
Hast du dazu eine Beschreibung?

**Reply by sentinel1111** [reply]
Hi,

Eine Beschreibung habe ich nicht.
Ich mache das mit den "Flash Download Tool" oder auch direkt mit den Tasmota WEBInstaller aber installiere natürlich nicht Tasmota sondern die OpenDTUfactory.bin

**Reply by skippermeister** [reply]
Hallo @sentinel1111, Du hast die Erstinstallation auf Deinem Board über USB gemacht. Funktioniert es danach per OTA ? Achtung nur die fimware.bin, nicht firmware.factory.bin per OTA nehmen.

Hier die Command Sequence für ein ESP32 (mein eigenes Board my_very_special_board) um das firmware.bin file per esptool hochzuladen (bei mir z.B. über COM4):

esptool --port COM4 --baud 921600 --chip esp32 --before default_reset --after hard_reset write_flash --flash_mode dout --flash_freq 40m --flash_size detect 0x10000 opendtu-onbattery-my_very_special_board.bin

und dies für die firmware.factory.bin

esptool --port COM4 --baud 921600 --chip esp32 --before default_reset --after hard_reset write_flash --flash_mode dout --flash_freq 40m --flash_size detect 0x0 opendtu-onbattery-my_very_special_board.factory.bin

**Reply by skippermeister** [reply]
Hallo @iroc-fabi, funktioniert es bei Dir ? Wird das Meanwell Netzteil erkannt ?
Was für eine Batterie hast du? Du hast anscheinend ein 24 Volt System.
Was für ein BMS hast Du ? Oder nutzt Du ein VictronSmartShunt per VE.Direct ?  
Nutze auf jeden Fall die neue Version von heute

**Reply by iroc-fabi** [reply]
Hallo @skippermeister,
ich nutze ein 24V System mit dem Akku von Redodo. Der hat ein eingebautes BMS mit Bluetooth. An die Daten komme ich leider nicht per openDTU, deswegen ist noch ein SmartShunt verbaut. 
Deine Version konnte ich leider noch nicht testen, weil ich es nicht geschafft hatte die Version auf das Fusionboard zu installieren. :-(

**Reply by iroc-fabi** [reply]
Ich hab jetzt hiermit versucht das neuste Firmware von Dir aufzuspielen.
Funktioniert aber auch nicht :-(
https://3printd-solution.eu/Fusion-openDTU-Anleitung-Inbetriebnahme

![image](https://github.com/user-attachments/assets/f524bc7b-19e2-42c8-8762-79d3dc1218c6)

**Reply by iroc-fabi** [reply]
Ich korrigiere, Firmware ist jetzt drauf :-)

**Reply by iroc-fabi** [reply]
Ich hatte die alte config Datei hochgeladen. WLAN etc wurde alles übernommen.
Danach VE.Direct aktiviert und den SmartShunt eingerichtet.
Zum Schluss noch die alte pinmapping hochgeladen.
Jetzt geht nichts mehr  komme nicht mehr rauf. Auch ausstecken bringt nicht.

**Reply by sentinel1111** [reply]
@skippermeister Ja,nach der Erstinstallation über usb funktioniert OTA,mehrmals ausprobiert.

**Reply by skippermeister** [reply]
Dann poste mal Dein pinmapping file. Wie hast Du das erstellt? Das pinmapping file der originalen OpenDTU-onBattery ist mit meinem pinmapping file vom Format nicht kompatibel.

**Reply by skippermeister** [reply]
An welchen GPIOs hängt bei Dir welches Gerät dran?
Ich habe zwar kein FusionBoard, aber bei mir läuft die Version störungsfrei.

**Reply by skippermeister** [reply]
Am USB Port kannst Du den gesamten Bootvorgang beobachten (z.B. mit Putty). Gerne poste den log für den Zeitraum der ersten 5min nach dem Reset der CPU. Dann kann ich den analysieren.

**Reply by iroc-fabi** [reply]
Mit dieser pinmapping Datei wurde mit der normalen openDTU on Battery Version der SmartShunt erkannt
[pin_mapping.json](https://github.com/user-attachments/files/16347525/pin_mapping.json)
Da fehlt dann aber wohl sowieso noch ein Eintrag für das Ladegerät.

**Reply by iroc-fabi** [reply]
Daten vom Ladegerät werden empfangen. Leider wird die maximale Ausgangsleistung immer wieder überschrieben.
![image](https://github.com/user-attachments/assets/e46d27a3-15dc-4b15-a2e9-8417d61f6cda)

**Reply by skippermeister** [reply]
@iroc-fabi : Versuche mal dieses Pinmapping file. Ich habe das eben erstellt.

[pin_mapping.json](https://github.com/user-attachments/files/16350546/pin_mapping.json)

Beachte: Mein Pinmapping file hat eine geänderte Struktur und Semantik.
Hinweis: Wenn Du im Pinmapping fille reinschaust, wirst Du einen Fehler finden. Unter Victron steht "rs323_rx" bzw. "rs323_tx" usw. Das wird in der aktuellen Version so gelesen, also nicht ändern. Ich werde es in der nächsten Version ändern in "rs232_rx" bzw. "rs232_tx", usw.

Ich habe aber noch in meiner Version ein Problem gefunden. Die LiveView der vom SmartShunt überwachten Batterie ist an meine Änderung noch nicht angepasst. Was wird vom SmartShunt überhaupt bei Dir angezeigt ? Ich kann das selbst nicht testen, da ich kein SmartShunt habe. Ich werde versuchen die Anpassung für die LiveView des SmartShunt am Wochenende einzubauen.

**Reply by iroc-fabi** [reply]
Wie müssen die Jumper eingestellt werden?

**Reply by iroc-fabi** [reply]
Lädt leider nicht. 
![IMG_8330](https://github.com/user-attachments/assets/629f3270-bf1e-4ae4-a2ae-3a9ea60bf38f)

**Reply by iroc-fabi** [reply]
Laden funktioniert leider immer noch nicht.
Vielleicht liegt es an der Jumpereinstellung

smartshunt meldet diese Werte.
![image](https://github.com/user-attachments/assets/c9ccbdae-c80d-4559-9f91-684bc8f6d69f)

**Reply by iroc-fabi** [reply]
das passt auch noch nicht. irgendwie geht aktuell wohl nur 48V System
![image](https://github.com/user-attachments/assets/86fb2a4d-4f89-459b-87d8-373f901d16b7)

**Reply by skippermeister** [reply]
Dein Inverter produziert ca. 90 Watt. Das Vorzeichen im Powermeter muss geändert werden. Dein Haus verbraucht ca. -970 Watt. Und vom Grid werden -1060Watt bezogen. Das kann nicht sein. Das Vorzeichen muss geändert werden. 
Nach der Änderung schalte mal im WebGui LiveView das Netzteil manuell an.
Ein automatisches Laden wird bei Dir mit dem SmartShunt nicht funktionieren. Der SmartShunt meldet nicht, dass der Akku sofort geladen werden muss.
Im Normalbetrieb läd doch tagsüber der Victron MPPT den Akku.

**Reply by iroc-fabi** [reply]
Der inverter ist nicht nur die drei HMS. Dazu gehören auch zwei größere SMA WR. 

Ladegerät manuell einschalten funktioniert nicht.

**Reply by iroc-fabi** [reply]
Dir Vorzeichen kann ich  nicht ändern, die kommen so vom SMA Zähler.

**Reply by skippermeister** [reply]
Das habe ich falsch interpretiert. Ich wusste nicht dass Du noch eine große Anlage hast. Ich hatte mich über den großen Power Export to Grid gewundert. Die Werte stimmen also bei Dir

**Reply by skippermeister** [reply]
OpenDTU kennt ja nicht die Solarleistung Deiner SMA Anlage und nimmt daher an, dass Du einen negativen Verbrauch im Haus hast.

**Reply by iroc-fabi** [reply]
heute funktioniert es nicht mehr.

Switch to WiFi mode
20:18:25.438 > Setting Hostname... done
20:18:25.488 > Configuring WiFi STA using new credentials... done
20:18:25.549 > Configuring WiFi STA DHCP IP... done
20:18:25.618 > [PowerLimiter]announceStatus: disabled by configuration
20:18:25.680 > [ZeroExport] invalid inverter selection/configuration
20:18:25.740 > [MEANWELL] read parameter
20:18:25.798 > WiFi connected
20:18:25.859 > WiFi got ip: 192.168.0.34
20:18:25.918 > Network connected
20:18:25.980 > Connecting to MQTT...
20:18:26.035 > Connected to MQTT.
20:18:26.083 > Websocket: [/livedata][1] connect
20:18:26.143 > [MEANWELL] Failed to queue message for transmission
20:18:26.198 > [MEANWELL] Failed to queue message for transmission
20:18:26.265 > [MEANWELL] Failed to queue message for transmission
20:18:26.323 > [MEANWELL] Failed to queue message for transmission
20:18:26.375 > [MEANWELL] Failed to queue message for transmission
20:18:26.427 > [MEANWELL] Failed to queue message for transmission
20:18:26.490 > [MEANWELL] Failed to queue message for transmission
20:18:26.549 > [MEANWELL] Failed to queue message for transmission
20:18:26.606 > [MEANWELL] Failed to queue message for transmission
20:18:26.669 > [MEANWELL] Failed to queue message for transmission
20:18:26.726 > [MEANWELL] Failed to queue message for transmission
20:18:26.775 > [MEANWELL] Failed to queue message for transmission
20:18:26.823 > [MEANWELL] Failed to queue message for transmission
20:18:26.885 > [MEANWELL] Failed to queue message for transmission
20:18:26.983 > [MEANWELL] Failed to queue message for transmission
20:18:27.044 > [MEANWELL] Failed to queue message for transmission
20:18:27.103 > [MEANWELL] Failed to queue message for transmission
20:18:27.186 > [MEANWELL] Failed to queue message for transmission
20:18:27.294 > [MEANWELL] Failed to queue message for transmission
20:18:27.392 > [MEANWELL] Failed to queue message for transmission
20:18:27.493 > [MEANWELL] Failed to queue message for transmission
20:18:27.544 > [MEANWELL] Failed to queue message for transmission
20:18:27.604 > [MEANWELL] Failed to queue message for transmission
20:18:27.702 > [MEANWELL] SystemConfig: 00000100 00000001
20:18:27.751 > [MEANWELL] Failed to queue message for transmission
20:18:27.810 > [MEANWELL] Failed to queue message for transmission
20:18:27.904 > [MEANWELL] Failed to queue message for transmission
20:18:28.010 > [MEANWELL] Failed to queue message for transmission
20:18:28.109 > [MEANWELL] Failed to queue message for transmission
20:18:28.158 > [MEANWELL] done
20:18:28.226 > [MEANWELL] Failed to queue message for transmission
20:18:28.271 > [MEANWELL] Failed to queue message for transmission
20:18:28.322 > [MEANWELL] Failed to queue message for transmission
20:18:28.371 > [MEANWELL] Round trip 3002 ms
20:18:28.428 > [VE.Direct SmartShunt 10/9] checksum 0x2e != 0x00, invalid frame
20:18:28.479 > Fetch inverter: 116480156021
20:18:28.524 > [ZeroExport] invalid inverter selection/configuration
20:18:28.576 > Success
20:18:30.293 > [MEANWELL] Failed to queue message for transmission
20:18:30.360 > [MEANWELL] Failed to queue message for transmission
20:18:30.411 > [MEANWELL] Failed to queue message for transmission
20:18:30.472 > [MEANWELL] Round trip 3001 ms

**Reply by skippermeister** [reply]
Ich habe soeben eine neue Version hochgeladen. 

Beachte: Das Pinmapping für die Victron VE.Direct Bezeichnung habe ich von rs323_rx bzw. rs323_tx in rs232_rx und rs232_tx geändert. Die Änderung musst Du unbedingt im Pinmapping Json file übernehmen. Im Folder Docs/DeviceProfiles findest Du das korrigierte Pinmapping file FusionV2.json.

Im Setup Menü für die Battery sind für den Provider VictronSmartShunt zwei weitere Parameter hinzugekommen:
Empfohlene Ladeschlussspannung
Empfohlene Entladeschlussspannung
Da der SmartShunt diese Parameter vom BMS der Batterie nicht abfragen kann, gibt es hier nun die Möglichkeit diese einzutragen.
Der Meanwell Charger stellt sicher, dass die Batterie nicht weiter als die Ladeschlussspannung geladen wird. Hier also die Werte Deiner Batterie eintragen.

Im LiveView WebGUI Section AC-Ladegerät habe ich die Eingabe der minimalen und maximalen Spannungs- und Stromwerte korrigiert. Das sollte jetzt funktionieren. Wenn Das Ladegerät korrekt erkannt wurde, dann werden jetzt die unteren und oberen Grenzwerte in Abhängigkeit vom erkannten Ladegerätes überprüft.

Im Setup Menü des AC-Ladegerätes können die oberen und unteren Grenzwerte innerhalb der technischen Spezifikationen des erkannten Netzteils noch weiter begrenzt werden. Beispiel:
Meanwell NPB-750-24  21V bis 42 Volt Spannungsbereich (technische Spezifikation des Netzteils)
Einstellung im AC-Ladegrät Setup Menü:
Begrenzung untere Spannungswert 21.5V
Begrenzung oberer Spannungswert 28.8V
Das Ladegerät kann dann nicht mit Spannungen unterhalb von 21.5 Volt oder oberhalb von 28.8 Volt eingestellt werden.

**Reply by skippermeister** [reply]
Die Fehlermeldung Deines Logs deuten auf eine fehlerhafte oder gestörte Kommunikation zwischen dem Meanwell Charger und dem CAN Bus Interface des Fusion Boards hin. Wie lang ist das Kabel zum Meanwell Charger? Mein Kabel ist nur ca. 50cm lang. Ich bin nicht sicher, ob auf der Meanwell Charger Seite am CAN Bus ein Abschlusswiderstand ist (im Gerät). Ich meine nicht. Sollte Dein Kabel deutlich länger sein, könnte das eine Störungsquelle sein. Bei 50cm Kabellänge sind die Signalreflexionen gering bzw. vernachlässigbar.

**Reply by iroc-fabi** [reply]
In deiner pinmapping ist noch ein Fehler.
can0_rx steht auf 12 und muss 13.

![image](https://github.com/user-attachments/assets/ed53a412-49b3-4d94-9eef-25be894a6a1d)

Wieso schnellt die Anzahl der Schreibzugriffe so hoch? Kann man das begrenzen? einmal die Minute oder so?
Es gibt Ladegeräte die haben einen maximalen Schreibzugriff. Kann ich das erkennen?

Mir ist noch aufgefallen, dass die Werte vom Ladegerät nicht zu Homeassistent gesendet werden. SmartShunt ja.

**Reply by skippermeister** [reply]
@iroc-fabi 
Zunächst Danke für den Hinweis bzgl. can0_rx, werde ich gleich korrigieren und mit dem nächsten Upload sichtbar machen.

Ich sehe noch einen Fehler. Die Netzeingangsspannung steht auf 0.0V. Das Meanwell NPB-750-xx liefert diesen Wert nicht., Da sollte eigentlich 230 Volt stehen. Die größeren Ladegeräte NPB-1200 und NPB-1700 liefern diese Werte.
Korrigiere ich in der nächsten Version.

**Fragen meinerseits:**
Hast Du auf Ausgangsstrom 4.5A eingestellt ?
Was war denn bei Dir der Fehler, dass auf einmal nichts mehr funktionierte ?
Und was funktioniert jetzt alles bei dir ? Und was funktioniert noch nicht ?

**Jetzt zu Deinen Fragen:**
Welche Version hat das Handbuch deines Meanwell Ladegerätes ? Ab Handbuch Version 7 sollen de Meanwell NPB-xxx-xx Ladegeräte die Funktion zur Vermeidung von EEPROM Write beinhalten. Meine Software versucht das entsprechende Bit zu setzen. Du kannst das während Bootvorganges auch im Serial Console Log sehen, ob das Bit gesetzt werden kann.
Die Versionsnummer des Handbuches findest Du im Inhaltsverzeichnis unten rechts.

**Hinweis zum Disable Write to EEPROM**
Siehe auf Seite 53 im Handbuch:
High byte: 
Bit 10  EEP_OFF: Disable to write voltage and current parameters to EEPROM 
0= Write the voltage and current parameters into EEPROM in real time（Default)
1= Disable to write the voltage and current parameters into EEPROM in real time

In Deinem obigen log findest Du folgende Zeile
20:18:27.702 > [MEANWELL] SystemConfig: 00000**1**00 00000001
Das ist ein Befehl an das Ladegerät. Ich versuche das Bit 10 zu setzen. In der Antwort müsste dann erscheinen "EEPROM write disable: 1". Wenn da eine 0 steht, dann unterstützt Dein Ladegerät diese Funktionen noch nicht.
Die EEPROM Write Zyklen werden aber in jedem Fall hochgezählt. Wer ein neueres Ladegerät hat, der kann die Werte ignorieren.

Schnellen die Schreibzugriffe auch im Modus "Manuelle Ladung" hoch? Das sollte eigentlich nicht sein.
Wenn Du auf Automatik gestellt hast, dann kannst Du die EEPROM Writes durch erhöhen der "Hysterese des Zero Grid Ladereglers" im AC Ladegerät Setup Menü reduzieren. Ich habe bei mir einen Wert von 25 Watt eingegeben. Ich schreibe ca. 3000 Write pro Tag, das macht ca. 10 Jahre Lebenszeit. Ich untersuche die Technik des Ladegerätes und will bei mir das EEPROM (falls es auf der Platine sichtbar ist) gegen ein FRAM austauschen. Das FRAM hat bei der Schreibhäufigkeit eine Lebenserwartung von theoretisch weit über 100 Jahre.

Die daten des Meanwell Ladegerätes werden nicht nach HomeAssistant gesendet. Diese Funktion ist noch nicht implementiert. Ich meine, das ist auch nicht in der original OpenDTU-onBattery implementiert. Oder sehe ich das falsch? Ich habe es im Source Code nicht finden können.

**Reply by iroc-fabi** [reply]
Ich hatte das Kabel, mit dem man das Ladegerät einschaltet, also Pin 7+8 getrennt. Wusste nicht, dass das immer zusammen sein muss.
Deswegen kamen keine Daten an.

Leider regnet es nonstop, so kann ich die automatische Steuerung nach PV-Leistung nicht testen. Aber vielleicht klappt es morgen.

Leider steht in meinem Handbuch Version 6 :-(. Hatte es erst vor zwei Wochen bei Völkner gekauft.

Heute steht da schon wieder was mit 42V, warum?
![image](https://github.com/user-attachments/assets/142bae4c-1689-4d2d-bb55-64dbbf1aacac)

**Reply by iroc-fabi** [reply]
Welche Werte sollte ich hier einstellen?
![IMG_8375](https://github.com/user-attachments/assets/10504b50-e325-41fe-b366-a0325b102c4f)

**Reply by iroc-fabi** [reply]
Die Werte vom SmartShunt werden von openDTU übertragen 
![IMG_8376](https://github.com/user-attachments/assets/5d2fbc26-65f5-403c-8169-e57bc0f9bfba)

**Reply by iroc-fabi** [reply]
Die Angaben V Angaben stimmen immer noch nicht. -> 42V

![image](https://github.com/user-attachments/assets/120acd8d-fd39-4bc9-8b2e-31f9ecf387d8)

**Reply by iroc-fabi** [reply]
Was wird da geprüft? Wenn die Hoymiles WR Strom erzeugen, dann wird das Ladegerät eingeschaltet?
Oder wenn insgesamt Strom ins Netz eingespeist wird, dann wird das Ladegerät eingeschaltet? (Bei mir dann inkl. der großen SMA WR)
Was macht der Schalter genau?

![image](https://github.com/user-attachments/assets/aac12272-a2a4-4bf0-98e2-3bf66c82fbc3)

**Reply by skippermeister** [reply]
Ich lade gleich noch eine neue Version hoch. Kleine Korrekturen in den HASS Modulen.

Hast Du die minimale und maximale Ausgangsspannung im Setup Menü "AC Ladegerät" eingestellt. Der SmartShunt liefert diese nicht und das NPB-750-24 Ladegerät hat eine obere Maximalspannung von 42 Volt. Diese Spannung musst Du für Deine Batterie natürlich begrenzen.

Ich würde für Deine Batterie mit diesen Werten versuchen zu arbeiten:
AC Ladegerät Menü
Minimale Ausgangsspannung	21.6 Volt
Maximale Ausgangsspannung	28.8 Volt
Minimaler Ausgangsstrom		4.5A
Maximaler Ausgangsstrom		22.5A

Batterie Menü
Empfohlene Ladeschlussspannung	28.8 Volt
Empfohlene Entladeschlussspannung	21.6 Volt

DPL Menü
Start-Schwellwert	27.5 Volt	musst Du probieren, würde ich aber auf jeden Fall größer 24.8 Volt setzen
Stop-Schwellwert	23.2 Volt	musst Du auch probieren, auf jeden Fall deutlich größer als 21.6 Volt
Lastkorrekturfaktor	hängt von Deinen Innenwiderständen ab, wird für Fallback benötigt, wenn keine SoC Werte geliefert werden.

Der Paramater "Prüfe ob Inverter erzeugt solare Energie" hat folgende Funktion:
Wenn das Ladegerät auf "Automatik Ladung" steht (nur Solar Strom) wird das Ladegerät nur aktiviert wenn der Hoymiles Wechselrichter auch Strom produziert.
Der Automatik Modus stellt das Ladegerät auf die erzeugte Überschussleistung ein, sodass nur Solar Energie zum Laden verwendet wird.

Jedoch falls die Batterie ein "Full Charge Request" anfordert, dann schaltet das Ladegerät mit aktiviertem Parameter "Prüfe ob Inverter erzeugt solare Energie" nur ein und lädt die Batterie auf 100%, wenn der Hoymiles Wechselrichter Solar Strom produziert. 
Ist der Parameter deaktiviert, wird das Ladegerät auf jeden Fall mit seiner maximal eingestellten Leistung "Maximaler Ausgangsstrom" aktiviert und die Batterie wird auf 100% aufgeladen.

Da Du noch die große SMA Anlage hast, kann dieser Parameter deaktiviert werden.

Im Mode "Manuelle Ladung" wird der Parameter "Prüfe ob Inverter erzeugt solare Energie" ignoriert.
Der Lade-Algorithmus reagiert aber auf den "Immediate Charge Request" und auf den "Full Charge Request" und lädt bei 

	Immediate Charge request" die Batterie auf den Wert des Parameters "Laden bei SoC beenden" (Batterie Menü) 
	bzw. bei "Full Charge Request" auf 100% auf.

Das Schütz den Akku insbesondere im Winterhalbjahr vor einer Tiefentladung.

Der Parameter "Hysterese des Zero Grid Ladereglers" ist der Wert um wieviel die berechnete Leistung des neu berechneten Sollwertes für die Ladeleistung sich mindestens ändern muss, bis dieser an das Ladegerät gesendet wird. Damit kann die Anzahl der EEPROM writes reduziert werden.

**Reply by skippermeister** [reply]
@iroc-fabi
Hattest Du noch weitere Tests durchgeführt? Im letzten Release gab es ein Problem mit dem neu integrierten SPIPortManager. Dieser ist nun behoben.

Eeine neues Release mit Korrekturen im SPIPortManager habe ich soeben hochgeladen.

**Reply by iroc-fabi** [reply]
So, bin jetzt aus dem Urlaub. Wollte grad einen HMS 400 zusätzlich zu den beiden HMS 1600 installieren.
Da kommt dann die Meldung, dass nur drei WR unterstützt werden.
Ist das so?

**Reply by skippermeister** [reply]
@iroc-fabi 
ich habe in der Configuration.h Datei den Parameter `INV_MAX_COUNT` auf 3 reduziert. Meine Version benötigt für die Pylontech Batterie mehr RAM Speicher. Da ich während der Entwicklung meiner Version RAM Resourcen Probleme hatte, habe ich den Parameter von 5 auf 3 reduziert.
3 Inverter müssen aber funktionieren. In meiner Testumgebung habe ich 3 Inverter konfiguriert.
Falls Du 4 benötigst, müsstest Du Dir eine spezielle Version für bis zu 4 Inverter kompilieren.
